server/src/main/java/org/opensearch/rest/action/admin/cluster/RestNodesStatsAction.java

```diff
@@ -209,6 +229,10 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
             nodesStatsRequest.indices().includeOnlyTopIndexingPressureMetrics(request.paramAsBoolean("top", false));
         }
 
+        // If no levels are passed in this results in an empty array.
+        String[] levels = Strings.splitStringByCommaToArray(request.param("level"));
+        nodesStatsRequest.indices().setLevels(levels);
+
```

Comment:
For 1), @jainankitk and I had a discussion about this. Basically, we want to make levels accessible to ImmutableCacheStatsHolder at creation time, so they must be present in NodesStatsRequest. That way we can aggregate at creation time, rather than having to aggregate during ToXContent. This also means each node only has the load of aggregating its own stats, rather than every node serializing the whole tree, sending it to the coordinator, and the coordinator having to aggregate each one. Ankit opened an issue around this [here](https://github.com/opensearch-project/OpenSearch/issues/13340). 

For 2), see [this](https://github.com/opensearch-project/OpenSearch/pull/13237#discussion_r1578228740) comment below, but I think CommonStatsFlags is used for more than just indices